### PR TITLE
feat: uniswap-forks; Added Pangolin

### PR DESCRIPTION
**Pull request by robinbernon**

### CODE REVIEW

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    return BigDecimal.fromString("0.25");
+  }
+  getProtocolFeeToOff(): BigDecimal {
+    return BigDecimal.fromString("0");
+  }
+  getLPFeeToOff(): BigDecimal {
+    return BigDecimal.fromString("0.3");
+  }
+  getFeeOnOff(): string {
+    return FeeSwitch.ON;
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts:63]: Did you ask in the discord about this? I was not able to verify in the docs that the protocol fee is on and what the  percentage is.

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      "0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e", // USDC
+      "0xa7d7079b0fead91f3e65f86e8915cb59c1a4c664", // USDC.e
+      "0xc7198437980c041c805a1edcba50c1ce5db95118", // USDT.e
+      "0x9702230a8ea53601f5cd2dc00fdbc13d4df4a8c7", // USDT
+      "0xde3a24028580884448a5397872046a019649b084", // USDT old
+    ];
+  }
+  getStableOraclePools(): string[] { // TODO - Need to change these values as they currently incorrect!
+    return [
+      "0xb4e16d0168e52d35cacd2c6185b44281ec28c9dc", // USDC/wETH created 10008355
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts:97]: USDC/Avax? The non-stable should match your reference token.

File: subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { BigInt } from "@graphprotocol/graph-ts";
+
+////////////////////
+//////Versions//////
+////////////////////
+
+export const PROTOCOL_SUBGRAPH_VERSION = "1.1.8";
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts:7]: You can set the subgraph version to 1.0.0

File: subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+export const PROTOCOL_METHODOLOGY_VERSION = "1.0.0";
+export const PROTOCOL_NAME = "Pangolin";
+export const PROTOCOL_SLUG = "Pangolin";
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts:10]: Slug should be `pangolin`

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+  "deployment": "PANGOLIN",
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.json:2]: Should be `PANGOLIN_AVALANCHE`. Also, you should add the Pangolin avalanche configurations to configurations/configurations.ts

**Review by steegecs — COMMENTED** [review]
Nice work! 

Is this subgraph currently working with the deployment scripts I showed you? (e.g. `npm run build`)

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts (hunk 1 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts:63]: Ah Im not sure, I think they said in the whitepapers they were actually going to start with the fees off with the option to turn it on later so perhaps I should switch this to off

File: subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts (hunk 3 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts]: Thanks will do this before deploying final subgraph 👍

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts (hunk 2 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts]: Yep I realised this also during the chat we had yesterday - I've updated this now

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.json (hunk 5 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.json]: Thanks have changed to this

File: subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts (hunk 4 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/src/common/constants.ts]: Thanks have changed it to this

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts (hunk 1 above)
Comment (steegecs) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts:63]: I see. Maybe double check. It looks like they have not updated with SetFeeTo when I look at the Factory contract history. So whatever it starts as should be what it is now. https://snowtrace.io/txs?a=0xefa94de7a4656d787667c749f7e1223d71e9fd88&p=2

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts (hunk 2 above)
Comment (steegecs) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts]: Great!

File: subgraphs/uniswap-forks/protocols/pangolin/src/mappings/masterchef/rewardMini.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    miniChefPool.poolAllocPoint,
+    event.params.allocPoint,
+    MasterChef.MINICHEF
+  );
+  miniChefPool.poolAllocPoint = event.params.allocPoint;
+  miniChefPool.save();
+}
+
+// Update the total emissions rate of rewards for the masterchef contract.
+export function handleLogRewardPerSecond(event: LogRewardPerSecond): void {
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/src/mappings/masterchef/rewardMini.ts:92]: Set the reward interval to TIMESTAMP in the configurations.ts.

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  getProtocolFeeToOff(): BigDecimal {
+    return BigDecimal.fromString("0");
+  }
+  getLPFeeToOff(): BigDecimal {
+    return BigDecimal.fromString("0.3");
+  }
+  getFeeOnOff(): string {
+    return FeeSwitch.OFF;
+  }
+  getRewardIntervalType(): string {
```
Comment (steegecs) [inline subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts:65]: RewardIntervalType.TIMESTAMP

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts (hunk 1 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts]: Yes you're right they haven't set a feeTo yet so I've turned this off

File: subgraphs/uniswap-forks/protocols/pangolin/src/mappings/masterchef/rewardMini.ts (hunk 6 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/src/mappings/masterchef/rewardMini.ts:92]: Have changed, thanks 👍

File: subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts (hunk 7 above)
Comment (robinbernon) [reply subgraphs/uniswap-forks/protocols/pangolin/config/deployments/pangolin-avalanche/configurations.ts:65]: Have changed, thanks 👍

**Review by steegecs — APPROVED** [review]
LGTM

**Pull request merged**